libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1313,3 +1313,20 @@ void RC_Channels::convert_options(const RC_Channel::aux_func_t old_option, const
         }
     }
 }
+
+bool RC_Channel::radio_input_changed()
+{
+    if (!rc().ignore_receiver()) {
```

Comment:
I wonder if we should add, "rc().has_had_rc_receiver() &&" to this line.

Reply:
An alternative implementation that would allow avoiding two calls to hal.rcin->read() would be to add a "bool rc_in_changed" member and then update it from within RC_Channel::update().  Something like below although it would be good to get opinions from other developers on whether this is useful or not.

```
// read input from hal.rcin or overrides
bool RC_Channel::update(void)
{
    const rc_in = hal.rcin->read(ch_in);
    rc_in_changed = radio_input_changed(rc_in);

    if (has_override() && !rc().ignore_overrides()) {
        radio_in = override_value;
    } else if (rc().has_had_rc_receiver() && !rc().ignore_receiver()) {
        radio_in = rc_in;
    } else {
        return false;
    }

    if (type_in == RC_CHANNEL_TYPE_RANGE) {
        control_in = pwm_to_range();
    } else {
        //RC_CHANNEL_TYPE_ANGLE
        control_in = pwm_to_angle();
    }

    return true;
}
```

Reply:
yes, two calls to hal.rcin->read() are not good.
I accept your suggestion.